core/src/org/javarosa/xpath/expr/XPathFuncExpr.java

```diff
@@ -145,6 +145,20 @@ public Object eval(DataInstance model, EvaluationContext evalContext) {
 
         Hashtable funcHandlers = evalContext.getFunctionHandlers();
 
+        for (int i = 0; i < args.length; i++) {
+            argVals[i] = args[i].eval(model, evalContext);
+        }
+
+        //check for custom handler, use this if it exists.
+        try {
+            IFunctionHandler handler = (IFunctionHandler) funcHandlers.get(name);
+            if (handler != null) {
+                return evalCustomFunction(handler, argVals, evalContext);
+            }
+        } catch(XPathArityException e){
+            // we matched the name but not the arg count.

```

Comment:
Please add tests to verify that you actually override an existing function. Try overriding the "true" function, having it take in an argument and return "1". This is a good test, because it also checks arity errors, which you don't test for currently.